### PR TITLE
Use absolute value for bouncy in collision

### PR DESCRIPTION
**Pull request by pecksin**
Currently, a node with negative bouncy does "nothing" except when it doesn't.

Examples:
* colliding with -80 does nothing
* colliding with -100 does "nothing" but fills the terminal with messages about avoiding an infinite loop
* colliding with -120 **freezes the game**

This patch makes collision use the absolute value, and masks all bouncy nodes as positive for old clients (updated to protocol 41 for 5.6.0).

This fixes the problem, does not conflict with any current valid use, and prepares the code to use negative values for a "controllable/uncontrollable" flag later (#11939).

## To do
Ready for Review.

## How to test

Create a world with negative bouncy nodes

* Old server w Old client:
   no bounce, and at -120 or so the game freezes
* New server w Old client:
   client sees all bouncy nodes as positive
* Any server w New client:
   client sees negative nodes but treats them as positive

### CODE REVIEW

**Comment by appgurueu** [comment]
`abs` seems arbitrary and overly permissive to me. I'd prefer it if an error was thrown at node registration.

**Comment by pecksin** [comment]
@appgurueu This is partly to prepare for #11939 where I want to use negative as an "uncontrollable bounce" flag.  Can you think of any problems this would cause as-is?

File: src/collision.cpp [hunk 1]
```diff
@@ -504,7 +504,7 @@ collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
 							d));
 
 			// Get bounce multiplier
-			float bounce = -(float)nearest_info.bouncy / 100.0f;
+			float bounce = -(float)abs(nearest_info.bouncy) / 100.0f;
```
Comment (sfan5) [inline src/collision.cpp:507]: Since the value is passed on you don't need the abs here.

**Comment by v-rob** [comment]
This makes sense.  I don't think it really matters as to whether it takes the absolute value, clamps it to zero, or gives an error; there's never going to be a situation where negative bounciness will make any sense.  Will review later.

**Comment by pecksin** [comment]
@SmallJoker 
>     1. Send `abs()` values to old clients (protocol_version < 40) in the node def to avoid old clients from freezes
I didn't realize this was an option. I will look into it immediately.

**Review by Df458 — APPROVED** [review]
Looks like it works, for both new and old clients. I haven't worked too much with bouncy nodes so I can't say with 100% certainty, but judging by the code and testing I believe this should work well.

**Pull request merged**